unicodetools/src/main/java/org/unicode/xml/UcdXML.java

```diff
@@ -0,0 +1,825 @@
+            return Range.CJKUNIFIEDIDEOGRAPH;
+        }
+        if (gc.equals(UcdPropertyValues.General_Category_Values.Surrogate)) {
+            return Range.SURROGATE;
+        }
+        if (gc.equals(UcdPropertyValues.General_Category_Values.Private_Use)) {
+            return Range.CHARACTER;
+        }
+        if (NChar.equals(UcdPropertyValues.Binary.Yes.getShortName())) {
+            return Range.NONCHARACTER;
+        }
+        return Range.RESERVED;
+    }
+
+    private static int getLastCodepointInGroup(
+            AttributeResolver attributeResolver, int lowCodepoint, int highCodepoint) {
+        String blk = attributeResolver.getAttributeValue(UcdProperty.Block, lowCodepoint);
+        for (int codepoint = lowCodepoint; codepoint <= highCodepoint; codepoint++) {
+            if (!blk.equals(attributeResolver.getAttributeValue(UcdProperty.Block, codepoint))) {
+                return codepoint - 1;
+            }
+            if (codepoint == 0x20 - 1 // put the C0 controls in their own group
+                    || codepoint == 0xa0 - 1 // put the C0 controls in their own group
+                    || codepoint == 0x1160 - 1 // split the jamos into three groups
+                    || codepoint == 0x11a8 - 1 // split the jamos into three groups
+                    || codepoint == 0x1f1e6 - 1 // put the regional indicators in their own group
+            ) {
+                return codepoint;
+            }
+        }
+        return highCodepoint;
+    }
+
+    private static AttributesImpl getAttributes(
+            VersionInfo version,
+            AttributeResolver attributeResolver,
+            int codepoint,
+            UCDXMLOUTPUTRANGE outputRange) {
+        AttributesImpl attributes = new AttributesImpl();
+        attributes.addAttribute(
+                NAMESPACE, "cp", "cp", "CDATA", attributeResolver.getHexString(codepoint));
+
+        for (UcdPropertyDetail propDetail : UcdPropertyDetail.ucdxmlValues()) {
+            UcdProperty prop = propDetail.getUcdProperty();
+            if (version.compareTo(propDetail.getMinVersion()) >= 0
+                    && (propDetail.getMaxVersion() == null
+                            || version.compareTo(propDetail.getMaxVersion()) < 0)) {
+                String attrValue = attributeResolver.getAttributeValue(prop, codepoint);
+                boolean isAttributeIncluded =
+                        getIsAttributeIncluded(
+                                attrValue,
+                                attributeResolver.isUnihanAttributeRange(codepoint),
+                                propDetail,
+                                prop,
+                                outputRange);
+                if (isAttributeIncluded) {
+                    String propName = prop.getShortName();
+                    if (propName.startsWith("cjk")) {
+                        propName = propName.substring(2);
+                    }
```

Comment:
FYI: that substring should be the same as the prop.getLongName()

Reply:
There is no getLongName() in UcdProperty.java.

Reply:
maybe it's called something else? getName()? seems like there should be a function for each one of the two main aliases.

Reply:
Changed to:
`                        if (propName.startsWith("cjk")) {
                            propName = prop.getNames().getAllNames().get(1);
                        }
`